CAAPH panic when a HRP should be deleted

**What steps did you take and what happened:**
- create a cluster with a label that matches a hcp
- cluster gets deployed and so does the hrp
- remove the label so that the hcp should no longer match
- caaph crashes with panic (logs below)

<details>
  <summary>logs</summary>
  
  ```
I0627 14:48:42.192680       1 controller.go:228] "Starting workers" controller="helmchartproxy" controllerGroup="addons.cluster.x-k8s.io" controllerKind="HelmChartProxy" worker count=10
I0627 14:48:42.192770       1 controller.go:228] "Starting workers" controller="helmreleaseproxy" controllerGroup="addons.cluster.x-k8s.io" controllerKind="HelmReleaseProxy" worker count=10
I0627 14:48:42.450134       1 controller.go:118] "Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference" controller="helmreleaseproxy" controllerGroup="addons.cluster.x-k8s.io" controllerKind="HelmReleaseProxy" HelmReleaseProxy="daphne/openstack-cinder-csi-satellite-dev-tw4lh" namespace="daphne" name="openstack-cinder-csi-satellite-dev-tw4lh" reconcileID=80fff3d9-3315-4473-97ba-9bd6658543e8
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1f069ba]

goroutine 270 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:119 +0x1fa
panic({0x21bba80, 0x3cd1c80})
	runtime/panic.go:884 +0x212
sigs.k8s.io/cluster-api-addon-provider-helm/internal.generateHelmUninstallConfig(...)
	sigs.k8s.io/cluster-api-addon-provider-helm/internal/helm_operations.go:461
sigs.k8s.io/cluster-api-addon-provider-helm/internal.UninstallHelmRelease({0x290d608?, 0xc0007994d0?}, {0xc001028800?, _}, {{{0xc000680720, 0x7}, {0xc000680728, 0x6}, {0xc000680730, 0xd}, ...}, ...})
	sigs.k8s.io/cluster-api-addon-provider-helm/internal/helm_operations.go:482 +0x7a
sigs.k8s.io/cluster-api-addon-provider-helm/controllers/helmreleaseproxy.(*HelmReleaseProxyReconciler).reconcileDelete(0x29362f8?, {0x290d608, 0xc0007994d0}, 0xc000699440, {0xc001028800, 0x15da})
	sigs.k8s.io/cluster-api-addon-provider-helm/controllers/helmreleaseproxy/helmreleaseproxy_controller.go:259 +0x6c5
sigs.k8s.io/cluster-api-addon-provider-helm/controllers/helmreleaseproxy.(*HelmReleaseProxyReconciler).Reconcile(0xc000997020, {0x290d608, 0xc0007994d0}, {{{0xc000680700, 0x6}, {0xc0008338f0, 0x28}}})
	sigs.k8s.io/cluster-api-addon-provider-helm/controllers/helmreleaseproxy/helmreleaseproxy_controller.go:141 +0xaba
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x290d608?, {0x290d608?, 0xc0007994d0?}, {{{0xc000680700?, 0x20a6360?}, {0xc0008338f0?, 0x0?}}})
	sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:122 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0004b5a40, {0x290d560, 0xc000a17bc0}, {0x228af80?, 0xc00029e5e0?})
	sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:323 +0x38f
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0004b5a40, {0x290d560, 0xc000a17bc0})
	sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:274 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:235 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:231 +0x333

  ```
</details>



**What did you expect to happen:**
caaph should delete the helm installation and the hrp that corresponds to it

**Anything else you would like to add:**
- running manually a helm uninstall on the workload cluster fixes the issue
- i cannot get more detailed logs as there is no way to increase verbosity

**Environment:**

- Cluster API version: 1.4.2
- Cluster API Add-on Provider for Helm version: v0.1.0-alpha.7
- minikube/kind version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

/kind bug

--- a/internal/helm_operations.go
+++ b/internal/helm_operations.go
@@ -131,7 +131,6 @@ func InstallOrUpgradeHelmRelease(ctx context.Context, kubeconfig string, spec ad
 func generateHelmInstallConfig(actionConfig *helmAction.Configuration, helmOptions *addonsv1alpha1.HelmOptions) *helmAction.Install {
 	installClient := helmAction.NewInstall(actionConfig)
 	installClient.CreateNamespace = true
-
 	if helmOptions == nil {
 		return installClient
 	}
@@ -458,6 +457,10 @@ func ListHelmReleases(ctx context.Context, kubeconfig string, spec addonsv1alpha
 // generateHelmUninstallConfig generates default helm uninstall config using helmOptions specified in HCP CR spec.
 func generateHelmUninstallConfig(actionConfig *helmAction.Configuration, helmOptions *addonsv1alpha1.HelmOptions) *helmAction.Uninstall {
 	uninstallClient := helmAction.NewUninstall(actionConfig)
+	if helmOptions == nil {
+		return uninstallClient
+	}
+
 	uninstallClient.DisableHooks = helmOptions.DisableHooks
 	uninstallClient.Wait = helmOptions.Wait
 	if helmOptions.Timeout != nil {

